src/core/physics/collisions.rs

```diff
@@ -182,6 +219,62 @@ pub enum PhysicsError {
     BodyNotInitialized(String),
 }
 
+#[derive(Default, Clone)]
+pub struct PhysicsHooks;
+
+impl rapier::PhysicsHooks for PhysicsHooks {
+    fn filter_contact_pair(
+        &self,
+        _context: &rapier::PairFilterContext,
+    ) -> Option<rapier::SolverFlags> {
+        // No contact pair filtering hook currently implemented
+        Some(rapier::SolverFlags::COMPUTE_IMPULSES)
+    }
+
+    fn filter_intersection_pair(&self, _context: &rapier::PairFilterContext) -> bool {
+        // No intersection pair hook currently implemented
+        true
+    }
+
+    fn modify_solver_contacts(&self, context: &mut rapier::ContactModificationContext) {
+        // Determine if jump through modifiation is needed:
+        // (If one body is a player, and other is jump through tile.)
+
+        let collider1 = context.colliders.get(context.collider1).unwrap();
+        let body1 = context.bodies.get(context.rigid_body1.unwrap()).unwrap();
+        let collider2 = context.colliders.get(context.collider2).unwrap();
+        let body2 = context.bodies.get(context.rigid_body2.unwrap()).unwrap();
+
+        let mut jump_through_body: Option<&rapier::RigidBody> = None;
+        let mut other_body: Option<&rapier::RigidBody> = None;
+
+        // Determine which body is jump through collider, if any.
+        if collider1
+            .solver_groups()
+            .memberships
+            .intersects(SolverGroup::JUMP_THROUGH.bits().into())
+        {
+            jump_through_body = Some(body1);
+            other_body = Some(body2);
+        } else if collider2
+            .solver_groups()
+            .memberships
+            .intersects(SolverGroup::JUMP_THROUGH.bits().into())
+        {
+            jump_through_body = Some(body2);
+            other_body = Some(body1);
+        }
+
+        if jump_through_body.is_some() {
+            let other_body = other_body.unwrap();
+
+            if other_body.linvel().y > 0.0 {
```

Comment:
This isn't quite correct, as if body is already inside jump through and starts falling, it will start colliding and pop downwards a bit faster than expected, but not too worried about that yet.